### PR TITLE
[no squash] Minor graphics code fixes

### PR DESCRIPTION
**Pull request by sfan5**
ping @appgurueu

### CODE REVIEW

File: src/client/content_cao.cpp [hunk 1]
```diff
@@ -814,15 +814,21 @@ void GenericCAO::addToScene(ITextureSource *tsrc, scene::ISceneManager *smgr)
...
-		if (m_animated_meshnode)
-			m_animated_meshnode->getMesh()->setHardwareMappingHint(scene::EHM_STATIC);
+
+		if (m_animated_meshnode) {
+			auto *mesh = m_animated_meshnode->getMesh();
+			// skinning happens on the CPU
+			if (m_animated_meshnode->getJointCount() > 0)
+				mesh->setHardwareMappingHint(scene::EHM_STATIC, scene::EBT_VERTEX);
+			else
+				mesh->setHardwareMappingHint(scene::EHM_STREAM, scene::EBT_VERTEX);
```
Comment (grorp) [inline src/client/content_cao.cpp:829]: is this meant to be this way around?

Comment (sfan5) [reply src/client/content_cao.cpp:829]: no, it's the wrong way around

**Comment by wsor4035** [comment]
this currently does appear to fix https://github.com/minetest/minetest/issues/14999

**Comment by sfan5** [comment]
The circumstances are still a mystery to me but that's good to hear.

**Review by grorp — COMMENTED** [review]
Entity/player lighting works, both with and without shaders, both on Android and on desktop.

**Comment by grorp** [comment]
We have three different pipeline models now, that's confusing.

- Minetest shaders
- Real fixed pipeline on opengl (supports lighting stuff)
- Fake fixed pipeline on ogles2 (Irrlicht shaders) (doesn't support lighting stuff)

Since we can't really use the fixed pipeline lighting stuff, we could just drop it from Irrlicht's materials altogether. Thoughts?

And/or we could force-enable Minetest shaders on ogles2.

File: irr/src/CMakeLists.txt [hunk 2]
```diff
@@ -506,6 +506,9 @@ target_link_libraries(IrrlichtMt PRIVATE
 if(WIN32)
 	target_compile_definitions(IrrlichtMt INTERFACE _IRR_WINDOWS_API_) # used in _IRR_DEBUG_BREAK_IF definition in a public header
 endif()
+if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+	target_compile_definitions(IrrlichtMt INTERFACE _DEBUG) # same
```
[the remark this answers is not in the document]
Comment (sfan5) [reply irr/src/CMakeLists.txt:510]: yeah, it just never caused any problems.

> Why not standard NDEBUG instead?

refactors welcome

File: irr/include/IReferenceCounted.h [hunk 3]
```diff
@@ -136,6 +136,7 @@ class IReferenceCounted
 		return ReferenceCounter;
 	}
 
+#ifdef _DEBUG
```
[the remark this answers is not in the document]
Comment (sfan5) [reply irr/include/IReferenceCounted.h:139]: it's entirely unrelated but I noticed `IReferenceCounted` was wasting 8 (or 4) bytes for a never-used pointer in release builds, so I fixed it.

**Comment by sfan5** [comment]
> Since we can't really use the fixed pipeline lighting stuff, we could just drop it from Irrlicht's materials altogether. There's not much left to do for that. Thoughts?

We should definitely get rid of `SMaterial.Lighting` entirely.
It's true by default but set to false almost everywhere.

> And/or we could force-enable Minetest shaders on ogles2.

In general we could. But not because of this bug, it's just a coincidence that it didn't happen with shaders on.

**Review by grorp — APPROVED** [review]
LGTM

**Pull request merged**